praw/models/reddit/emoji.py

```diff
@@ -0,0 +1,81 @@
+"""Provide the EMOJI class."""
+from ...const import API_PATH
+from ..listing.generator import ListingGenerator
+from .base import RedditBase
+from .redditor import Redditor
+
+
+class Emoji(RedditBase):
+    """An individual Emoji object."""
+
+    __hash__ = RedditBase.__hash__
+
+    @property
+    def mod(self):
+        """Provide an instance of :class:`.WikiPageModeration`."""
+        if self._mod is None:
+            self._mod = EmojiModeration(self)
+        return self._mod
+
+    def __init__(self, reddit, subreddit, name, revision=None, _data=None):
+        """Construct an instance of the WikiPage object.
+
+        :param revision: A specific revision ID to fetch. By default, fetches
+            the most recent revision.
+
+        """
+        self.name = name
+        self.subreddit = subreddit
+        super(Emoji, self).__init__(reddit, _data)
+        self._mod = None
+
+
+class EmojiModeration(object):
+    """Provides a set of moderation functions for a WikiPage."""
+
+    def __init__(self, wikipage):
+        """Create a WikiPageModeration instance.
+
+        :param wikipage: The wikipage to moderate.
+
+        """
+        self.emoji = emoji
+
+    def add(self, emoji_name, filepath):
```

Comment:
I wonder if this should be set up such that it's invoked like `subreddit.emoji.add` instead of on a particular instance of an emoji. Thoughts?

Reply:
I think this makes some sense from an object oriented perspective, but given the complexity with the S3 Lease I was thinking it made sense to have a more atomic Emoji element itself. I think you're right that subreddit.emoji.add would be more consistent with the code base, just want to minimize confusion.